fml/trace_event.cc

```diff
@@ -22,6 +23,27 @@ namespace tracing {
 
 #if TIMELINE_ENABLED
 
+namespace {
+AsciiTrie gWhitelist;
+
+inline void FlutterTimelineEvent(const char* label,
+                                 int64_t timestamp0,
+                                 int64_t timestamp1_or_async_id,
+                                 Dart_Timeline_Event_Type type,
+                                 intptr_t argument_count,
+                                 const char** argument_names,
+                                 const char** argument_values) {
+  if (gWhitelist.Query(label)) {
+    Dart_TimelineEvent(label, timestamp0, timestamp1_or_async_id, type,
+                       argument_count, argument_names, argument_values);
+  }
```

Comment:
I may be missing something, but it looks like if you don't specify the whitelist you won't get any tracing.  That seems undesirable to me.  

Could we create a fast-track here so that if the whitelist hasn't been specified we don't bother querying it and just trace everything?  Or perhaps another switch for it?

Reply:
In case it's not clear: it seems like it would be best to make sure that all trace events are sent today if you don't specify the whitelist, rather than no trace events being sent. Otherwise it will be very difficult to figure out which trace events you want to whitelist (e.g. which ones are available/interesting) while analyzing performance issues.

Reply:
That's how it works, no whitelist means everything is traced.  You can verify that in the unit tests, an empty trie always returns true.